### PR TITLE
api: Add gRPC status proto definition

### PR DESCRIPTION
**Pull request by tonya11en**
Adds a gRPC status proto definition, analogous to the HTTP status found in `http_status.proto`. Currently, this proto is not in use, but is needed for https://github.com/envoyproxy/envoy/issues/9658.

Risk Level: Low
Testing: n/a
Docs Changes: Proto documented
Release Notes: n/a

### CODE REVIEW

File: api/envoy/type/grpc/v2/grpc_status.proto [hunk 1]
```diff
@@ -0,0 +1,40 @@
+syntax = "proto3";
```
Comment (htuch) [inline api/envoy/type/grpc/v2/grpc_status.proto:1]: Where is this used or planned to be used?

File: api/envoy/type/grpc/v2/grpc_status.proto [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  INTERNAL = 13;
+  UNAVAILABLE = 14;
+  DATA_LOSS = 15;
+  UNAUTHENTICATED = 16;
+}
+
+// GRPC status.
+message GrpcStatus {
+  // Supplies GRPC response code.
+  Status status = 1 [(validate.rules).enum = {defined_only: true not_in: 0}];
```
Comment (htuch) [inline api/envoy/type/grpc/v2/grpc_status.proto:39]: Does this mean that it can't be OK?

File: api/envoy/type/grpc/v2/grpc_status.proto [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+import "udpa/annotations/migrate.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.type.grpc.v2";
+option java_outer_classname = "GrpcStatusProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_migrate).move_to_package = "envoy.type.grpc.v3";
+
+// [#protodoc-title: GRPC status codes]
```
Comment (htuch) [inline api/envoy/type/grpc/v2/grpc_status.proto:13]: Nit: s/GRPC/gRPC/ everywhere

File: api/envoy/type/grpc/v2/grpc_status.proto [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+// GRPC response codes supported.
+enum Status {
```
Comment (mattklein123) [inline api/envoy/type/grpc/v2/grpc_status.proto:16]: See @lizan comment in the other PR about not using an enum for this.

File: api/envoy/type/grpc/v2/grpc_status.proto (hunk 2 above)
Comment (mattklein123) [reply api/envoy/type/grpc/v2/grpc_status.proto:39]: I had the same question/comment in the other PR.

File: api/envoy/type/grpc/v2/grpc_status.proto (hunk 1 above)
Comment (mattklein123) [reply api/envoy/type/grpc/v2/grpc_status.proto:1]: See the linked PR in the description.

**Comment by tonya11en** [comment]
@mattklein123 I'm more than happy to drop this PR in favor of just using an integer, but my only concern would be readability of the configurations.

I'm fine either way, so feel free to just close this out if you'd prefer the integer specification.

**Comment by mattklein123** [comment]
> I'm fine either way, so feel free to just close this out if you'd prefer the integer specification.

My feeling is that we should use an integer per @lizan and just clearly document what the integer means with a link/reference to the code mappings. So yeah I will close.

**Pull request closed without merging**